source/geometry_model/ellipsoidal_chunk.cc

```diff
@@ -40,6 +42,326 @@ namespace aspect
+                   lon * 180/numbers::PI));
+            break;
+
+          case FILE_NONUNIFORM_GRID:
+            return static_cast<Functions::InterpolatedTensorProductGridData<2>*>(topography_data)->value (Point<2>(lat * 180/numbers::PI,
+                   lon * 180/numbers::PI));
+            break;
+
+          default:
+            AssertThrow(false,ExcMessage ("This topography function for enum with value " + boost::lexical_cast<std::string>(topo_type) + " has not been implemented."));
+            return 0;
+            break;
+        }
+      Assert(false,ExcMessage ("The code should never reach this part. Please check the code or contact the developer."));
+      return 0;
+    }
+
+    template <int dim>
+    void
+    EllipsoidalChunk<dim>::EllipsoidalChunkTopography::set_corners(std::vector<Point<2> > set_corners)
+    {
+      corners = set_corners;
+    }
+
+    template <int dim>
+    void
+    EllipsoidalChunk<dim>::EllipsoidalChunkTopography::set_topography_type(topo_types set_topo_type)
+    {
+      topo_type = set_topo_type;
+    }
+
+    template <int dim>
+    void
+    EllipsoidalChunk<dim>::EllipsoidalChunkTopography::set_point_lists(std::vector<std::vector<std::vector<double> > > set_point_lists)
+    {
+      point_lists = set_point_lists;
+    }
+
+    template <int dim>
+    void
+    EllipsoidalChunk<dim>::EllipsoidalChunkTopography::set_topography_values(std::vector<double> set_topography_values)
+    {
+      topography_values = set_topography_values;
+    }
+
+
+    template <int dim>
+    void
+    EllipsoidalChunk<dim>::EllipsoidalChunkTopography::set_topography_data (Function<2> *set_topography_data)
+    {
+      topography_data = set_topography_data;
+    }
+
+
+    template <int dim>
+    void
+    EllipsoidalChunk<dim>::EllipsoidalChunkTopography::set_grid_number_data_points(std::vector<double> &set_grid_number_data_points)
+    {
+      grid_number_data_points = set_grid_number_data_points;
+    }
+
+    template <int dim>
+    void
+    EllipsoidalChunk<dim>::EllipsoidalChunkTopography::set_topography_file(std::string set_topo_file_name)
+    {
+      topo_file = set_topo_file_name;
+    }
+
+    template <int dim>
+    topo_types
+    EllipsoidalChunk<dim>::EllipsoidalChunkTopography::get_topo_type () const
+    {
+      return topo_type;
+    }
+
+    template <int dim>
+    std_cxx11::array<std::pair<double,double>,2>
+    EllipsoidalChunk<dim>::EllipsoidalChunkTopography::get_endpoints () const
+    {
+      std_cxx11::array<std::pair<double,double>,2> endpoints;
+      // Minimal and maximal longitude
+      endpoints[0] = std::make_pair (corners[1][0]*numbers::PI/180, corners[0][0]*numbers::PI/180);
+      // Minimal and maximal latitude
+      endpoints[1] = std::make_pair (corners[3][1]*numbers::PI/180, corners[0][1]*numbers::PI/180);
+
+#ifdef DEBUG
+      for (unsigned int i=0; i<2; i++)
+        Assert (endpoints[i].first < endpoints[i].second,

```

Comment:
No need for the `#ifdef`: in optimized mode, the `Assert` is just replaced by an empty code block, and the compiler will remove the `for` loop.